guacamole-ext/src/main/resources/org/apache/guacamole/protocols/ssh.json

```diff
@@ -83,6 +83,11 @@
                     "name"    : "backspace",
                     "type"    : "ENUM",
                     "options" : [ "", "127", "8" ]
+                },
+                {
+                    "name"  : "terminal-type",
+                    "type"    : "ENUM",
+                    "options" : [ "xterm", "xterm-256color", "vt220", "vt100", "ansi", "linux" ]
```

Comment:
@mike-jumper can comment more on this, but I'm not sure that `xterm` and `xterm-256color` should be included in this list, yet.  There's a JIRA issue, 277 [1] , that deals with providing xterm compatibility, and it isn't completed, yet, AFAIK.  The other values should be fine, and I'm not sure if there are more we can add to the list.

[1] - https://issues.apache.org/jira/browse/GUACAMOLE-277

Reply:
As long as they're not the default (the server-side terminal emulator still uses "linux" if this parameter is omitted), it's probably OK. In fact, recent changes adding xterm-specific terminal codes will not have any effect unless "xterm" or "xterm-256color" are used.

I'm personally OK with this as long as it's documented in the manual that xterm and xterm-256color are not fully supported yet.